### PR TITLE
Feat/tab controller 2 orientation try 3

### PR DESCRIPTION
**Pull request by M-i-k-e-l**
## Description
TabController2 - add support for orientation change.
Changed the key in `TabBar` like we've discussed, unfortunately this change still required many changes because the key is changed inside `TabBar` and not on `TabBar` (not by the user), and so the hooks are **not** recreated.
Following each commit will make it easier.
Testing with both scrollable (i.e. phone) and non scrollable (i.e. high resolution tablet) will allow you to see the bugs fixed in the commits.

## Changelog
TabController2 - add support for orientation change.

### CODE REVIEW

File: src/components/tabController2/TabBarItem.tsx [hunk 1]
```diff
@@ -183,11 +183,19 @@ export default function TabBarItem({
     };
   });
 
+  const constantWidthStyle = useAnimatedStyle(() => {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/TabBarItem.tsx:186]: Fixed

File: src/components/tabController2/TabBarItem.tsx [hunk 2]
```diff
@@ -183,11 +183,19 @@ export default function TabBarItem({
...
+    if (itemWidth.current) {
+      return {flex: 0, width: itemWidth.current};
+    }
+
+    return {};
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/TabBarItem.tsx:191]: That was required by `useAnimatedStyle`, removed.

File: src/components/tabController2/useScrollToItem.ts [hunk 3]
```diff
@@ -106,19 +116,33 @@ const useScrollToItem = <T extends ScrollToSupportedViews>(props: ScrollToItemPr
   // const contentWidth = _.sum(itemsWidths);
   // TODO: const scrollEnabled = contentWidth.current > containerWidth;
 
+  useEffect(() => {
+    if (onReset && shouldReset.current) {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/useScrollToItem.ts:120]: Not sure I understood you, but...
At first the reset was done in `TabBar` first and then I've made changes in `useScrollToItem`, if I remember correctly there was a problem with the timing in the `useEffect` of a fixed width item, and I had to trigger the reset from `useScrollToItem` to make sure no data was lost.

File: src/components/tabController2/index.tsx [hunk 4]
```diff
@@ -56,9 +57,24 @@ function TabController({
   carouselPageWidth,
   children
 }: PropsWithChildren<TabControllerProps>) {
+  const [screenWidth, setScreenWidth] = useState<number>(Constants.windowWidth);
+  const orientation = useRef<orientations>(Constants.orientation);
+  useEffect(() => {
+    const onOrientationChange = () => {
+      if (orientation.current !== Constants.orientation) {
```
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/index.tsx:64]: Seems like there's an even after the orientation has changed (occurred on iPhone 8, iPad Pro 4th gen did not have it).

File: src/components/tabController2/useScrollToItem.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/tabController2/useScrollToItem.ts:120]: I meant that the `reset` functionality will be written in `useScrollToItem`
You'll just invoke it from TabBar right before or after (depends on the timing you need) you set the new orientation key.

you already have this useEffect
```
useEffect(() => {
    // @ts-expect-error TODO: fix forwardRef Statics
    if (tabBar.current?.isScrollEnabled()) {
      focusIndex(currentPage.value);
    }
  }, [containerWidth]);
```

I think it'll make the code clearer cause there'll be a single place where we handle the orientation change

File: src/components/tabController2/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/index.tsx:64]: On iPhone8 I see a second event and this `orientation.current !== Constants.orientation` is false.

File: src/components/tabController2/useScrollToItem.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply src/components/tabController2/useScrollToItem.ts]: Done

File: src/components/tabController2/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/tabController2/index.tsx:64]: Got ya.. 
Ok, let's leave it like this for now

**Review by ethanshar — APPROVED** [review]

**Pull request merged**